### ISSUE TITLE
Math parser / evaluator

### ISSUE THREAD

**Issue by alexanderadam** [johnjansen/crystal-libraries-needed#113]
I'm just curious:
is there any Shard for evaluating math formulas (like [dentaku](https://github.com/rubysolo/dentaku))?

**Comment by RX14** [comment]
There's `execute_infix` in my [`rpn.cr`](https://github.com/RX14/rpn.cr).

It doesn't support variables yet, though i'd definitely be open a PR.

**Comment by alexanderadam** [comment]
Wonderful! I guess I can close this then!
Thank you for responding so fast :+1:

**Comment by straight-shoota** [comment]
[Crinja](https://github.com/straight-shoota/crinja/) is a full-fledged template engine, but it's also capable of evaluating math expressions. Variables are obviously supported, as well as many more complex features. 
```cr
require "crinja"
Crinja.new.evaluate "1 + x", {"x" => 3} # => 4
```

There is no explicit mode to dumb it down to only allow math expressions, but individual features can be disabled in the config, creating a sandboxing mode.

**Comment by alexanderadam** [comment]
Thank you for the advice. I didn't thought of using template engines for evaling.
But when I needed a math evaluator, I needed it to be as 'dumb' as possible, to rule out misuse. So it should do mathematics and nothing else.

**Comment by RX14** [comment]
@alexanderadam if you need variables implemented just open an issue and i'll probably have a crack at it

**Comment by alexanderadam** [comment]
@RX14 I will. At the moment I was just wondering whether there is some math evaluator at all. But I will open an issue in case I will need variables for sure!

btw while we are on it: thank you (both) for your awesome libraries! :pray:
